### ISSUE TITLE
Rule no-console ignores `window.console`

### ISSUE THREAD

**Issue by fritx** [eslint/eslint#4826]
https://github.com/eslint/eslint/blob/master/lib%2Frules%2Fno-console.js

```js
         "MemberExpression": function(node) {

            if (node.object.name === "console") {
                context.report(node, "Unexpected console statement.");
            }

        }
```

How about adding this support?
The fact is that some of my partners prefer to write `window.console.log`.
Or, I have to add a rule for myself?
~~Btw, I find no-alert is simillar:~~
Btw, I just found no-alert seems to be nicer: https://github.com/eslint/eslint/blob/master/lib%2Frules%2Fno-alert.js

Cheers

**Comment by eslintbot** [comment]
@fritx Thanks for the issue! If you're reporting a bug, please be sure to include:

1. The version of ESLint you are using (run `eslint -v`)
2. What you did (the source code and ESLint configuration)
3. The actual ESLint output complete with numbers
4. What you expected to happen instead

Requesting a new rule? Please see [Proposing a New Rule](http://eslint.org/docs/developer-guide/contributing/new-rules) for instructions.

**Comment by michaelficarra** [comment]
How do you suggest we determine that `window` is the global object?

**Comment by fritx** [comment]
Thanks @michaelficarra 
I'm new to eslint, but it seems the rule no-alert is okay.
Currently, I wrote an internal rule and it works in our project.
Not sure is it right :sweat_smile: 

```js
         "MemberExpression": function(node) {

            var name = getPropertyName(node)  // added line, borrowed from no-alert

            if (name === "console") {
                context.report(node, "Unexpected console statement.");
            }

        }
```

**Comment by michaelficarra** [comment]
Make sure you copy `isGlobalThisReferenceOrGlobalWindow`.

**Comment by fritx** [comment]
Thanks. I will try my best soon to comprehend the logic in no-alert ;)

**Comment by nzakas** [comment]
@eslint/eslint-team thoughts on this?

**Comment by mysticatea** [comment]
I wrote a logic to catch a specified global variable for no-eval.
I can extract the logic to AST utils, then use it in every rule to prevent a use of specified global variables.

**Comment by xjamundx** [comment]
Ha I always use this as a workaround instead of an inline comment when I want `console.log`. Probably a good idea to add it if possible.

**Comment by kaicataldo** [comment]
Just to be clear - the accepted change is for the no-console rule to warn for `window.console`, correct?

**Comment by ilyavolodin** [comment]
@kaicataldo That's correct. Currently it only catches `console.` it should also be catching `window.console.` as well. We also need to verify that `window` is a global variable, and hasn't been overridden.

**Comment by nzakas** [comment]
We should probably allow configuration of the global object so you can specify window or something else.

**Comment by EricHenry** [comment]
Hey guys, I helped out with the no-console rule before. If no one else is working on this issue, I'd like to help. Do you guys have a good recommendation on how to start?  I think I only scratched the surface on figuring out how aslant actually works last time. Any guidance would be helpful. Thanks!

**Comment by nzakas** [comment]
@EricHenry sure, we'd love the help. I think what we're looking at is a configuration that looks like this:

```
no-console: ["error", {
    "allow": ["error"],
    "disallowGlobals": [ "window", "self" ]
}]
```

**Comment by kaicataldo** [comment]
@EricHenry Any specific questions you have or things you've had trouble with in getting started working on this?

**Comment by EricHenry** [comment]
@kaicataldo After re-reading the issue, I am thinking my implementation is probably wrong. Right now I am only looking for references to 'window' or 'self', but realized that I have to determine if those are references to the global scope.

Thanks to @fritx  and @mysticatea I am looking at the no-alert rule and the and the no-eval rule because both have that functionality. I know that I could copy the code, but I am trying to understand how it is being done before that. That is my biggest hurdle right now, understanding how to determine the global scope from the AST.

**Comment by kaicataldo** [comment]
I'm closing this issues since it has been open since January and does not have anyone from the team or the community willing to step forward and own the work to complete to it.
